lib/logflare/context_cache.ex

```diff
@@ -32,43 +41,51 @@ defmodule Logflare.ContextCache do
         value
 
       {:ok, {:cached, value}} ->
-        index_keys(context, cache_key, value)
+        # already cached don't re-index
         value
     end
   end
 
-  def bust_keys(values) do
-    {:ok, keys} = Cachex.keys(@cache)
+  @doc """
+  This function is called from the CacheBuster process when a new record comes in from the Postgres
+  write-ahead log. The WAL contains records. From those records the CacheBuster picks out
+  primary keys.
 
-    total =
-      Enum.count(keys, fn {token, cache_key} = key ->
-        with true <- token in values,
-             {context, _} = token,
-             context_cache = cache_name(context),
-             {:ok, true} <- Cachex.del(context_cache, cache_key) do
-          Cachex.del(@cache, key)
-        end
+  The records ARE the keys in the reverse cache (the ContextCache).
 
-        true
-      end)
+  We must:
+   - Find the key by the record primary key
+   - Delete the reverse cache entry
+   - Delete the cache entry for that context cache e.g. `Logflare.Users.Cache`
+  """
 
-    :telemetry.execute(
-      [:logflare, :context_cache, :busted],
-      %{count: total},
-      %{}
-    )
+  def bust_keys(values) do
+    {:ok, keys} = Cachex.keys(@cache)
+
+    Enum.each(keys, fn {token, phash_key} = key ->
+      with true <- token in values,
+           {context, _} = token,
+           context_cache = cache_name(context),
+           {:ok, cache_key} = Cachex.get(@cache, key),
+           {:ok, true} <- Cachex.del(context_cache, cache_key) do
+        Cachex.del(@cache, key)
+      end
+    end)
 
     {:ok, :busted}
   end
 
   def bust_keys(context, id), do: bust_keys([{context, id}])
 
-  defp index_keys(context, cache_key, value) do
-    keys_key = {{context, select_key(value)}, cache_key}
+  @spec cache_name(atom()) :: atom()
+  def cache_name(context) do
+    Module.concat(context, Cache)
+  end
 
-    {:ok, key} = Cachex.get(@cache, keys_key)
+  defp index_keys(context, cache_key, value) do
+    keys_key = {{context, select_key(value)}, :erlang.phash2(cache_key)}
```

Comment:
What will happen if there will be conflict?

Reply:
I think there is likely enough info here that it won't be. 

`{{Logflare.Users, 2334}, 12342335}` and phash2 is a combo of the function name and all arguments.

Reply:
And what if there will be something? What will be the impact? I think that will mean that the cache will be invalidated to early, isn't it? But we probably also should ad TTL to the cache entries just to be sure that we evict not needed entries that are super old.

Reply:
If there was a conflict it would override the cached results with the wrong e.g. User actually prior to hitting this bust code.

Reply:
Well no it would reverse index the wrong key. So yeah it would end up busting the wrong record which is fine. And yeah I will add a TTL to all of it just in case.